src/Cli.Tests/StringLogger.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Cli.Tests;
+
+/// <summary>
+/// Creates a logger that can be used in test methods to verify logging behavior
+/// by capturing the messages and making them available for verification.
+/// </summary>
+class StringLogger : ILogger
+{
+    public List<string> Messages { get; } = new();
+
+    public IDisposable BeginScope<TState>(TState state)
+    {
+        return new Mock<IDisposable>().Object;
```

Comment:
Lets Use the 'using' statement to dispose of the object returned by 'BeginScope'.

public IDisposable BeginScope<TState>(TState state)
{
    using (var scope = new Mock<IDisposable>().Object)
    {
        return scope;
    }
}

Reply:
@aaronpowell, that's correct at least from what I've been able to dig up from other implementations of `BeginScope()` with no functionality. (We don't currently use it anywhere but is required to be implemented by the interface contract). 

Microsoft.Extension.Logging has a few examples which implement BeginScope. One of those being [EventSourceLogger](https://github.com/dotnet/runtime/blob/03608ee819b977f6dd1edbf8f29bdefd510e1761/src/libraries/Microsoft.Extensions.Logging.EventSource/src/EventSourceLogger.cs#LL121C2-L125C43).
For loggers that don't do anything specific in `BeginScope(TState state)`, .net implemented a basic non-functioning scope, [NullScope](https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/Extensions/Logging/NullScope.cs), implementing IDisposable which we could probably utilize to clean this up in a future pr.

--------------

Microsoft Docs example of a custom[ logging provider show](https://learn.microsoft.com/en-us/dotnet/core/extensions/custom-logging-provider#create-the-custom-logger):
```csharp
public IDisposable? BeginScope<TState>(TState state) where TState : notnull => default!;
```
but that [doesn't work i](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/nullable-warnings?f1url=%3FappId%3Droslyn%26k%3Dk(CS8633)#mismatch-in-nullability-declaration)n our code due to a mismatch in nullability declaration error.